File: PR/Tema5/Ejercicio_10_11_12/src/equipofutbol/Empleado.java
package equipofutbol;

public class Empleado implements Nombrable {
    private String nombre;
    private double sueldo;
    private double dinero;

    public Empleado(String n, double s) {
        this.nombre = n;
        this.sueldo = s;
    }

    public double getSueldo() {
        return sueldo;
    }

    public double getDinero() {
        return dinero;
    }

    @Override
    public String getNombre() {
        return nombre;
    }

    public void cobrar(double cantidad) throws CabreoException {
        if (cantidad < this.sueldo) {
            this.sueldo += cantidad;
            throw new CabreoException(
                    "Al empleado " + this.nombre + " no se le han pagado " + (this.sueldo - cantidad) + " euros");
        } else {
            this.sueldo += cantidad;
        }
    }
}

File: DAW2/EC/T1/Test2/6/Empleado.js
export class Empleado {
  _dinero = 1;

  constructor(nombre, apellido, salario) {
    this._nombre = nombre;
    this._apellido = apellido;
    this._salario = salario;
  }

  get nombre() {
    return this._nombre;
  }

  get apellido() {
    return this._apellido;
  }

  get salario() {
    return this._salario;
  }

  get dinero() {
    return this._dinero;
  }

  set dinero(newDinero) {
    this._dinero = newDinero;
  }
}
